From a15c58354131bf379e9a570d69057083fe1799cc Mon Sep 17 00:00:00 2001
From: Jonh Wendell <jonh.wendell@redhat.com>
Date: Fri, 22 Jan 2021 00:08:07 +0100
Subject: [envoy] Do not download envoy-centos image (#234)

* Do not download envoy-centos image

We do not use it.

Also, change proxy SHA to match maistra proxy.

* Change proxy image to be centos based

And change default app sidecar to be centos based too.

Make `make docker` work again (#251)

- Point the Base URL to ours instead of upstream
- Download envoy-centos binary (Reverts
  https://github.com/maistra/istio/pull/234/commits/bb004bd823640ea15e747e21bbb7c0d706a6f9e3)
- Bump proxy SHA
---
 pilot/docker/Dockerfile.proxyv2              | 26 ++++++--------------
 pkg/test/framework/components/echo/config.go |  2 +-
 prow/lib.sh                                  |  6 ++---
 3 files changed, 12 insertions(+), 22 deletions(-)

diff --git a/pilot/docker/Dockerfile.proxyv2 b/pilot/docker/Dockerfile.proxyv2
index 02cded141b..7680d71b74 100644
--- a/pilot/docker/Dockerfile.proxyv2
+++ b/pilot/docker/Dockerfile.proxyv2
@@ -1,22 +1,9 @@
-# BASE_DISTRIBUTION is used to switch between the old base distribution and distroless base images
-ARG BASE_DISTRIBUTION=debug
+FROM centos:8
 
-# Version is the base image version from the TLD Makefile
-ARG BASE_VERSION=latest
-
-# The following section is used as base image if BASE_DISTRIBUTION=debug
-FROM gcr.io/istio-release/base:${BASE_VERSION} as debug
-
-# The following section is used as base image if BASE_DISTRIBUTION=distroless
-# This image is a custom built debian11 distroless image with multiarchitecture support.
-# It is built on the base distroless image, with iptables binary and libraries added
-# The source can be found at https://github.com/istio/distroless/tree/iptables
-# This version is from commit 20b20ec36f4621830ff0bfdec519577074df47ca.
-FROM gcr.io/istio-release/iptables@sha256:bae9287d64be13179b7bc794ec3db26bd5c5fe3fb591c484992366314c9a7d3d as distroless
-
-# This will build the final image based on either debug or distroless from above
-# hadolint ignore=DL3006
-FROM ${BASE_DISTRIBUTION:-debug}
+# hadolint ignore=DL3039,DL3041
+RUN dnf -y upgrade --refresh && \
+    dnf -y install iptables iproute openssl && \
+    dnf -y clean all
 
 WORKDIR /
 
@@ -28,6 +15,9 @@ ARG SIDECAR=envoy
 COPY envoy_bootstrap.json /var/lib/istio/envoy/envoy_bootstrap_tmpl.json
 COPY gcp_envoy_bootstrap.json /var/lib/istio/envoy/gcp_envoy_bootstrap_tmpl.json
 
+RUN useradd -m --uid 1337 istio-proxy
+RUN chown -R istio-proxy /var/lib/istio
+
 # Install Envoy.
 ARG TARGETARCH
 COPY ${TARGETARCH:-amd64}/${SIDECAR} /usr/local/bin/${SIDECAR}
diff --git a/pkg/test/framework/components/echo/config.go b/pkg/test/framework/components/echo/config.go
index 0ce220cf06..3a93073636 100644
--- a/pkg/test/framework/components/echo/config.go
+++ b/pkg/test/framework/components/echo/config.go
@@ -49,7 +49,7 @@ const (
 	Centos7      VMDistro = "Centos7"
 	Centos8      VMDistro = "Centos8"
 
-	DefaultVMDistro = UbuntuBionic
+	DefaultVMDistro = Centos8
 )
 
 // Config defines the options for creating an Echo component.
diff --git a/prow/lib.sh b/prow/lib.sh
index 7aa381b940..be43b174f6 100755
--- a/prow/lib.sh
+++ b/prow/lib.sh
@@ -103,11 +103,11 @@ function build_images() {
   # Build just the images needed for tests
   targets="docker.pilot docker.proxyv2 "
 
-  # use ubuntu:bionic to test vms by default
-  nonDistrolessTargets="docker.app docker.app_sidecar_ubuntu_bionic "
+  # use centos:8 to test vms by default
+  nonDistrolessTargets="docker.app docker.app_sidecar_centos_8 "
   if [[ "${SELECT_TEST}" == "test.integration.pilot.kube" ]]; then
     nonDistrolessTargets+="docker.app_sidecar_ubuntu_xenial docker.app_sidecar_ubuntu_focal docker.app_sidecar_ubuntu_bionic "
-    nonDistrolessTargets+="docker.app_sidecar_debian_9 docker.app_sidecar_debian_10 docker.app_sidecar_centos_7 docker.app_sidecar_centos_8 "
+    nonDistrolessTargets+="docker.app_sidecar_debian_9 docker.app_sidecar_debian_10 docker.app_sidecar_centos_7 docker.app_sidecar_ubuntu_bionic "
   fi
   targets+="docker.operator "
   targets+="docker.install-cni "
-- 
2.31.1

